### ISSUE TITLE
As LeGo, we would like additional fields to be added to the Study setup form in Sequencescape, so users don’t have to input data twice

### ISSUE THREAD

**Issue by harrietc52** [sanger/sequencescape#4443]
**User story**
As LeGo, we would like additional fields to be added to the Study setup form in Sequencescape, so users don’t have to input data twice (once in SS and again in eDAM).

**Who are the primary contacts for this story**
@harrietc52 

**Who is the nominated tester for UAT**
Odion A

**Acceptance criteria**
To be considered successful the solution must allow:
- [ ] the list of fields from the Google Sheet are added to the Study creation form
- [ ] specific fields show/ hide depending on another field, as specified in the Google Sheet
- [ ] each field has a “Hint"

**Dependencies**
This story is blocked by the following dependencies:
- #<issue_no.>
- sanger/<repo>#<issue_no.>

References
This story has a non-blocking relationship with:
- #<issue_no.>
- sanger/<repo>#<issue_no.>
 
**Additional context**
Questions for PSD
- Check `Study Title` and `Study Description` are already added in SS study setup form 

Questions for eDAM developer (Odion)
- For the fields which are `Show if`. Are each of those fields required when shown, or optional?
- Does `moratorium_publication_period` only need to show if `is_moratoria_attached` is “Yes”?
- Does `moratorium_publication_period` need “Yes” and “No”, or could it just be a dropdown of numbers 1-12, with 12 the default selected?
- What input is required for "Known EGA Dataset Accession numbers for this study”. What type is it? Is it required?
- What is the “Hint” text for each field? Please can it be added to the Google Sheet
- Separated `non_sanger_dac_name` and `non_sanger_dac_email` into two separate fields. Is that okay?
- Is `approved_by_lego` okay? A “yes”/ “No” input? Is the text description okay, or would you like anything adding? Would you like this to be nearer the start of the form? If “No”, can the user still submit the Study in Sequencescape?
- Would you like these questions to be in their own group, within the Study setup page? Where exactly would you like them to go on the form (at the bottom/ top etc?)
- Are all these questions only required if the Studies release strategy is EGA, or ENA, or Both (ie would these questions only want to appear if EGA/ENA checkbox is selected in Sequencescape set up form?)

MLWH story:
Questions for eDAM developer (Odion)
- Confirm the `field` names? i.e check all `field`s in Google Sheet are agreed with eDAM as this is what they will end up being on `study` table in MLWH. 
- Is “Yes” “No”, okay for those fields in MLWH, or would you prefer booleans?
- For `include_collaborators_in_the_daa`, would those collaborators be created as it currently does, in eDAM when you pull in the study data?
- Could `data_access_requests` and `data_access_requests_email` be combined, to just be one field `data_access_requests_email`? If `data_access_requests` asks if the email is `datasharing@sanger`, or if not, input another email? Assume email requires `.ac.uk`?
- Can `sign_off_by_head_of_policy` and `sign_off_alternative_email` become one field? Which defaults to Sarion B email, or the manually entered email? What is this Sarion’s email address, and is it at a non-personal / role level? (incase Sarion B ever left the institute)

Fields to be added:
[Google Sheet](https://docs.google.com/spreadsheets/d/1Ya6YyBuoGNBV-6xFLfGHc1WuLjeVLjSlCOMQC6VCUhQ/edit?usp=sharing)

**Comment by harrietc52** [comment]
Q1: Is the Steps 1 & 2 in the Lucid chart correct? i.e The first time the Study is created in the eDAM2 internal system (and therefore when the DAA is created) is when the data from the study tables in the MLWH is pulled into eDAM2 internal? (OR is the Study first created first via eDAM, and then updated when data is pulled in?): 
> So studies are not created in eDAM at all, they are pulled from the MLWH. The RMs then have to go into eDAM to fill in the other information in the form for a particular study. 


2. Is this for both “Standard” and “Non-Standard” studies? Do both Standard and Non-Standard studies follow the same process? Do both Standard and Non-Standard have the same DAA questions?
> I'm not sure about standard and non standard studies. Laura feel free to jump in if you're aware of this process.

3. Currently in Sequencescape Study set up, there is a Study Name field. Is this linked to the eDAM field Study Title? Would you like the “Hint” for this field in eDAM adding to Sequencescape?: 
> Regarding the Study Name field, I think so, but I can't say for sure. This is the query we run to get the studies from the MLWH: "select id_study_lims, description, accession_number, study_title, faculty_sponsor from study where data_release_strategy='managed' ", so it depends on whether the Study Name field in Sequencescape is inserted into the study table in the MLWH as study_title. Regarding the Hint, you can leave that out.

4. Currently in Sequencescape Study set up, under ENA specific questions there is a Title (study_study_title) and Study Description (study_description). Is the Study Description in eDAM different (not ENA specific), and if so, shall a new field for “Study Description" to the Sequencescape Study set up page? What would you like the text to show, and what would you like the field in MLWH to be?: 
> The Study Description in eDAM comes from the 'description' field in the study table in the MLWH. I think it's the ENA specific question, but I can't say for sure.

5. The same field name collabs refers to both name an institution. Should these fields be seperated to collabs_name and collabs_institution?: 
> No, leave as it is (see response to next question below)

6. There is a note in "eDAM DAA setup questions” - “Users must be able to add multiple collabs”. How would you like this to be persisted in the study table in MLWH? If there is more than one collaborator?
> Might be best to save it as a json array, so something like this: [ { collabName: "", collabInstitute: "" }, { collabName: "", collabInstitute: "" } ]. We can have a call to discuss this.

7. Is want_pub_delay needed? For example, could this field be removed and only the publish_delay field used, and set to 12 as default?: 
> Yeah, it can be removed. 

8. What input is required for "Known EGA Dataset Accession numbers for this study”. What type is it? Is it required? It is missing from "eDAM DAA setup questions” Google Sheet below
> No input is required for this because the RMs don't enter this information. These are the datasets linked to the study, eDAM pulls these datasets from the EGA archive and displays them on this question.

9. What is the “Hint” text for each field? Please can it be added to the Google Sheet (see eDAM fields below) Do not all fields have a “Hint”?:
> Not all fields have hints.

10. Can we separate non_sanger_dac_notes into two serpate fields non_sanger_dac_name and non_sanger_dac_email?: > Would you like this changed Laura?

11. Could dao_send_to_giselle and dao_send_to be combined, to just be one field dao_send_to? By default the email is datasharing@sanger, or if not, input another email? Assume email requires .ac.uk? Then don't have a users name in a field: > Yeah, I'm happy with that, if Laura is happy with it as well

12. Can ez1_to_approve be removed? And sanger_approve_by become one field? Which defaults to Sarion B email, or the manually entered email? Could ez1’s address be replaces with a non-personal / role level? (incase Sarion B ever left the institute). Also, in "eDAM DAA setup questions” doc it says Display when option 1 is selected for ez1_to_approve, is this correct, or should it display when “No” is selected for ez1_to_approve?: 
> Yeah, you can merge the fields, if Laura is happy with that. Regarding replacing Sarion's email with a role, we can discuss how that would work. You're right, it should be Option 2, I've updated the google sheet.

13. In "eDAM DAA setup questions” doc it says Display when option 1 is selected for in_edam , is this correct, or should it display when “No” is selected for in_edam ?:
> It should display when "No" is selected, I've updated the google sheet.

14. Is approved_by_lego okay? A “yes”/ “No” input? Is the text description okay, or would you like anything adding? e.g. “Confirm this study has been approved by LeGo and added to eDAM2 internal”. Would you like this to be nearer the start of the form? If “No”, can the user still submit the Study in Sequencescape?. 
> Could you clarify what role this question plays when answered?

15. Would you like all these questions to be in their own group, within the Study setup page? Where exactly would you like them to go on the form (at the bottom/ top etc?): 
> Laura, do you have a preference regarding the location of the questions.

16. Are all these questions only required if the Studies release strategy is EGA, or ENA, or Both (ie would these questions only want to appear if EGA/ENA checkbox is selected in Sequencescape set up form?): 
> Referring this to Laura J

**Comment by harrietc52** [comment]
Todo:
- [ ] Add Laura’s questions to Flowchart
- [ ] Add Odions answers to Google Sheet

**Comment by harrietc52** [comment]
Questions (All)

- eDAM Study Title; In SS Study creation, would you like the `Study name` or `Study Title` field to be used in eDAM? (currently using `Study Title`, which is “ENA” - should it say ENA if it required for both ENA & EGA?)
- eDAM Study Description; In SS Study creation, is it still okay for the `Study Description (ENA)` to be used in eDAM? - should it say ENA if it required for both ENA & EGA?

Questions (Odion)
- `include_collabs`, is this needed? Either `collabs` will have data or it will not? And therefore `collabs` always show?
- Is there any use in adding the field `ega_dataset_accession_numbers` to SS Study creation page or the MLWH?

**Comment by harrietc52** [comment]
- eDAM section want to appear if study is “Standard” (to check language), this section wants to be shown by default?
- eDAM fields want to be in a separate section in the Study set up page
- eDAM fields want to be read-only, after creation
- there wants to be a message on the Study view page saying that if the user would like to update any eDAM fields, to do so in eDAM directly

**Comment by harrietc52** [comment]
###  Questions (outdated)

Currently in Sequencescape Study set up, there is a `Study Name` field. Is this linked to the eDAM field `Study Title`? Would you like the “Hint” for this field in eDAM adding to Sequencescape?
> Regarding the Study Name field, I think so, but I can't say for sure. This is the query we run to get the studies from the MLWH: "select id_study_lims, description, accession_number, study_title, faculty_sponsor from study where data_release_strategy='managed' ", so it depends on whether the Study Name field in Sequencescape is inserted into the study table in the MLWH as study_title. Regarding the Hint, you can leave that out.

Currently in Sequencescape Study set up, under ENA specific questions there is a `Title` (`study_study_title`) and `Study Description` (`study_description`). Is the `Study Description` in eDAM different (not ENA specific), and if so, shall a new field for “Study Description" to the Sequencescape Study set up page? What would you like the text to show, and what would you like the field in MLWH to be?: 
> The Study Description in eDAM comes from the 'description' field in the study table in the MLWH. I think it's the ENA specific question, but I can't say for sure.

The same field name `collabs` refers to both name an institution. Should these fields be seperated to `collabs_name` and `collabs_institution`?: 
> No, leave as it is (see response to next question below)

There is a note in "eDAM DAA setup questions” - “Users must be able to add multiple collabs”. How would you like this to be persisted in the `study` table in MLWH? If there is more than one collaborator?
> Might be best to save it as a json array, so something like this: [ { collabName: "", collabInstitute: "" }, { collabName: "", collabInstitute: "" } ]. We can have a call to discuss this.

Is `want_pub_delay` needed? For example, could this field be removed and only the `publish_delay` field used, and set to 12 as default?: 
> Yeah, it can be removed. 

`ega_dataset_accession_numbers` What input is required for "Known EGA Dataset Accession numbers for this study”. What type is it? Is it required? It is missing from "eDAM DAA setup questions” Google Sheet below
> No input is required for this because the RMs don't enter this information. These are the datasets linked to the study, eDAM pulls these datasets from the EGA archive and displays them on this question.

What is the “Hint” text for each field? Please can it be added to the Google Sheet (see eDAM fields below) Do not all fields have a “Hint”?:
> Not all fields have hints.

Can we separate `non_sanger_dac_notes` into two serpate fields `non_sanger_dac_name` and `non_sanger_dac_email`?: 
> Yes this would be fine, it might make it a bit clearer and would be useful to ensure we have both the name and email address.

Could `dao_send_to_giselle` and `dao_send_to` be combined, to just be one field `dao_send_to`? By default the email is `datasharing@sanger`, or if not, input another email? Assume email requires `.ac.uk`? Then don't have a users name in a field. 
> Yeah, I'm happy with that, if Laura is happy with it as well
> As far as I know this should always be the datasharing mailbox, so yes it should default to this but have an option for something else just in case of odd situations.

Can `ez1_to_approve` be removed? And `sanger_approve_by` become one field? Which defaults to Sarion B email, or the manually entered email? Could `ez1`’s address be replaces with a non-personal / role level? (incase Sarion B ever left the institute). Also, in "eDAM DAA setup questions” doc it says Display when option 1 is selected for `ez1_to_approve`, is this correct, or should it display when “No” is selected for `ez1_to_approve`?: 
> Yeah, you can merge the fields, if Laura is happy with that. Regarding replacing Sarion's email with a role, we can discuss how that would work. You're right, it should be Option 2, I've updated the google sheet.
> As long as the approvals still go to Sarion then this is fine. If she left then I guess we would just update this to the email address of whoever took over doing approvals from her.

 In "eDAM DAA setup questions” doc it says Display when option 1 is selected for `in_edam `, is this correct, or should it display when “No” is selected for `in_edam `?
> It should display when "No" is selected, I've updated the google sheet.

Is `approved_by_lego` okay? A “yes”/ “No” input? Is the text description okay, or would you like anything adding? e.g.  “Confirm this study has been approved by LeGo and added to eDAM2 internal”. Would you like this to be nearer the start of the form? If “No”, can the user still submit the Study in Sequencescape?. 
> Could you clarify what role this question plays when answered?

Would you like all these questions to be in their own group, within the Study setup page? Where exactly would you like them to go on the form (at the bottom/ top etc?): 
>  If they could appears in SS in the same order as they appear in eDAM2 that would be good I think. So this would be below the study title and description sections.

Are all these questions only required if the Studies release strategy is EGA, or ENA, or Both (ie would these questions only want to appear if EGA/ENA checkbox is selected in Sequencescape set up form?): 
> These questions are only needed for managed access data held in EGA (not ENA), and only those using the standard WSI DAA (so if they’re using a different DAA like the DDD or CGP ones for example, then the questions aren’t needed).

**Comment by ah44aah** [comment]
Would it be possible to have a "help" button next to the questions related to eDAM, which would contain information regarding how to fill in the question/where to find the information needed?

**Comment by harrietc52** [comment]
**PSD questions:**

- How do new/edit/mange pages work. Is code shared?
- Check when this study data is sent to the MLWH? on “activation” or when study is submitted?
- Study Name vs Title - what populates what MLWH fields

**Comment by harrietc52** [comment]
**Ali questions:**
- Are any SSRs involved in the testing of this?
- Please check the ‘label’ for each question in the here ([eDAM questions to update in Sequencescape Study](https://docs.google.com/spreadsheets/d/1L2LQt5vROosUXbXBfHpWFdzgTr5_-XQNOEat68kOLwg/edit?usp=sharing)). e.g. does the Purpose question want the `EGA requirement` label, if it is not just EGA who use this information?

**Comment by TWJW-SANGER** [comment]
Closing and archiving as this will be superseded by Sapio integration.
